### ISSUE TITLE
Unable to run tests using specific trait via vstest.console /TestCaseFilter

### ISSUE THREAD

**Issue by rgusman** [xunit/xunit#180]
I was wondering if it is possible to run tests of a particular trait using vstest.console.exe? I am currently trying to execute tests with a specific trait during the build process but instead of only executing the tests with the trait specified, all tests are executed.

The command line that I am running looks similar to this;
 vstest.console.exe "Test.dll" /settings:"test.Xunit.runsettings" /Platform:x64 /TestCaseFilter:"Category=BuildTest" /Framework:Framework45 /UseVsixExtensions:true

**Comment by bquarmby** [comment]
Speaking of the Visual Studio test console, now that the xunit VSIX is retired, will it still be possible to run vstest.console.exe using the NuGet based Visual Studio adapter?

**Comment by bradwilson** [comment]
I honestly don't know the answer to this question.

If the Visual Studio console runner supports the same discovery logic that Visual Studio does, then it should work without incident. But frankly, I don't use either one, so ... experiment away and report back. :)

**Comment by bquarmby** [comment]
After some quick testing, the NuGet package works if the assemblies are copied to the extensions folder below vstest.console.exe. But I haven't been able to work out how to use them directly from the test project output folder like Visual Studio does. Wasn't as simple as making it the working directory.

**Comment by pmiossec** [comment]
@bquarmby I think I successfully did it and explained it on [stackoverflow](http://stackoverflow.com/a/27844550/717372).

The point is to use the `/TestAdapterPath` parameters to specify the path toward the xunit adapter.

You should use a command line looking like that:

    "C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe" c:\path\to\your\assembly.to.test.dll /TestAdapterPath:".\packages\xunit.runner.visualstudio.0.99.9-build1021\build\_common\"

**Comment by bquarmby** [comment]
@pmiossec Nice one, that's it. I've upvoted your answer on stackoverflow.

Because the xunit adapter files are copied to the test project output, the command can be reduced to something like:
D:\Solution\Project.Tests\bin\Debug>"C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\IDE\CommonExtensions
\Microsoft\TestWindow\vstest.console.exe" Project.Tests.dll /TestAdapterPath:.

That way it's not tied to a particular version of the NuGet package.

**Comment by onovotny** [comment]
The PR is here: https://github.com/xunit/visualstudio.xunit/pull/75

**Comment by bradwilson** [comment]
Fixed by xunit/visualstudio.xunit#75.

**Comment by onovotny** [comment]
That is a current limitation in the way the traits are built/discovered. We need to know the full list of possible traits in all assemblies but they discovery can happen in parallel.

The best solution for now is to have a dummy test that uses a trait you want to filter on in each assembly.

**Comment by htuomola** [comment]
I did guess that that's the workaround but was hoping for a better solution. Thanks for the fast reply anyway. Also, now this behavior is at least kind of documented in here.

**Comment by bradwilson** [comment]
@htuomola Please open an new issue so it can be fixed.

Thanks.

**Comment by htuomola** [comment]
Sure, will do. 
Also, I noticed that with xUnit, /TestCaseFilter="Category!=Foo" runs only test which have Category defined and it is not Foo. With MsTest also tests that do not contain Category at all are ran. I guess I can/should open a new issue for that too?

**Comment by bradwilson** [comment]
Yes please :)

**Comment by htuomola** [comment]
Sorry, my last comment about the non-existing traits/categories not getting executed was incorrect. That functionality works as expected.
